# HG changeset patch
# User SSlivenko
# Parent  69c63fe72e96093c9bb12aaff0597262ce231c72
SC0017729229 Некорректное заполнение тега <doc-date> в пакетах обмена с 1С ERP. TH16 ( все кроме doc-type 6 )

diff --git a/ibs/th/bge/1crn/subjects/trn-gd-doc.cls b/ibs/th/bge/1crn/subjects/trn-gd-doc.cls
--- a/ibs/th/bge/1crn/subjects/trn-gd-doc.cls
+++ b/ibs/th/bge/1crn/subjects/trn-gd-doc.cls
@@ -979,15 +979,57 @@ class ibs.th.bge.1crn.subjects.trn-gd-do
     
   end.
 
-  method private void StartDocTrn ():
+  /* method private void StartDocTrn ():
       
       InitSecTag =  "trn-gd-docs".
+      sw:start-element ("trn-gd-doc") .
+        write-data-element ("doc-code", buf_trn-doc.doc-code).
+        write-data-element ("doc-num", buf_trn-doc.doc-code).
+        write-data-element ("del-f", string (isDel, "1/0") ).
+        write-data-element ("doc-date", if buf_trn-doc.doc-date = ? then "" else substitute("&1T&2", iso-date(date(buf_trn-doc.doc-date)), string(buf_trn-doc.sys-time) ) ). 
+        write-data-element ("fact-date", if buf_trn-doc.fact-date = ? then "" else iso-date (datetime (buf_trn-doc.fact-date, buf_trn-doc.fact-time * 1000))).
+        write-data-element ("doc-type", v-doc-type-ERPRN).
+        if v-is-edo-return
+        then do :
+          write-data-element ("edi-return", "1").
+        end. 
+        write-data-element ("doc-rsn-code", (if buf_trn-doc.reason-code = ? then "0" else string-mth (buf_trn-doc.reason-code))).
+        write-data-element ("edoc-id", edoc-id).
+        write-data-element ("shift-date", if buf_trn-doc.shift-date = ? then "" else iso-date (date (buf_trn-doc.shift-date))).
+        write-data-element ("shift-num", if buf_trn-doc.shift-num = ? then "0" else string-mth (buf_trn-doc.shift-num)).
+        write-data-element ("contr-code", string-mth (buf_trn-doc.cli-code)).
+        write-data-element ("dog-code", string-mth (buf_trn-doc.contract-code)).
+        write-data-element ("operator", v-user-name ).
+        write-data-element ("psn-wrkr", string-mth (buf_trn-doc.wrkr)).
+        write-data-element ("psn-agnt", string-mth (buf_trn-doc.agnt)).
+        write-data-element ("psn-boss", string-mth (buf_trn-doc.boss)).
+  end. */
+  
+  
+  method private void StartDocTrn ():
+
+      define variable dtm as CHARACTER no-undo.
+      
+      InitSecTag =  "trn-gd-docs".
+      if buf_trn-doc.doc-date = ? then DO: 
+        dtm = "" .
+      END.
+      
+      else IF buf_trn-doc.doc-date <> ? then DO: 
+        find first c-trn-doc where c-trn-doc.doc-code = buf_trn-doc.doc-code NO-LOCK NO-ERROR.
+        IF NOT AVAILABLE c-trn-doc THEN DO: 
+           dtm = substitute("&1T&2", iso-date(date(buf_trn-doc.doc-date)), string(buf_trn-doc.sys-time) ).
+        END.
+        ELSE IF AVAILABLE c-trn-doc THEN DO:
+           dtm = substitute("&1T&2", iso-date(date(c-trn-doc.doc-date)), string(c-trn-doc.sys-time) ). 
+        END. 
+      END.
       
       sw:start-element ("trn-gd-doc") .
         write-data-element ("doc-code", buf_trn-doc.doc-code).
         write-data-element ("doc-num", buf_trn-doc.doc-code).
         write-data-element ("del-f", string (isDel, "1/0") ).
-        write-data-element ("doc-date", if buf_trn-doc.doc-date = ? then "" else iso-date (datetime (buf_trn-doc.doc-date, buf_trn-doc.real-time-create * 1000))).
+        write-data-element ("doc-date", dtm ). 
         write-data-element ("fact-date", if buf_trn-doc.fact-date = ? then "" else iso-date (datetime (buf_trn-doc.fact-date, buf_trn-doc.fact-time * 1000))).
         write-data-element ("doc-type", v-doc-type-ERPRN).
         if v-is-edo-return
@@ -1005,6 +1047,8 @@ class ibs.th.bge.1crn.subjects.trn-gd-do
         write-data-element ("psn-agnt", string-mth (buf_trn-doc.agnt)).
         write-data-element ("psn-boss", string-mth (buf_trn-doc.boss)).
   end.
+  
+  
 
   method private void EndDocTrn ():
     
